src/compiler/program.ts

```diff
@@ -903,11 +903,12 @@ namespace ts {
                     return;
                 }
 
+                const caseSensitive = host.useCaseSensitiveFileNames();
                 for (let i = 0, n = Math.min(commonPathComponents.length, sourcePathComponents.length); i < n; i++) {
-                    if (commonPathComponents[i] !== sourcePathComponents[i]) {
+                    if (caseSensitive ? commonPathComponents[i] !== sourcePathComponents[i] : commonPathComponents[i].toLocaleLowerCase() !== sourcePathComponents[i].toLocaleLowerCase()) {

```

Comment:
And now I'm using `getCanonicalFileName` again. :+1: 

Worked around the casing handling in the harness/emitter by using `getCanonicalFileName` on the path fragments when comparing fragments rather than on the whole initial path. This is similar to what we do within a few of our utility functions, so I suppose this use of `getCanonicalFileName` is to be expected, though I didn't expect it to be handled at first.